analyse_csv.py

```diff
@@ -0,0 +1,135 @@
+
+    dict_result['analysis_date'] = date_process
+
+    return dict_result
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--input_folder',
+                        help='The folder where the csv can be found',
+                        required=True)
+    parser.add_argument('--output_folder',
+                        help='The folder where the output will be made',
+                        required=True)
+    parser.add_argument('--date_process',
+                        default=TODAY)
+    parser.add_argument('--num_files',
+                        default='10')
+    parser.add_argument('--num_rows',
+                        default='500')
+    parser.add_argument('--num_cores',
+                        default='1')
+    parser.add_argument('--num_samples',
+                        default='0')
+    parser.add_argument('--output_mode',
+                        default='LIMITED')
+
+    args = parser.parse_args()
+
+    num_files = int(args.num_files)
+    num_rows = int(args.num_rows)
+    num_cores = int(args.num_cores)
+
+    input_folder = Path(args.input_folder)
+    output_folder = Path(args.output_folder)
+
+    analysis_name = os.path.basename(os.path.dirname(args.input_folder))
+    list_files = []
+    if input_folder.exists():
+        list_files = get_files(input_folder, n_sample=int(args.num_samples))
+    else:
+        logger.info("No file/folder found to analyze. Exiting...")
+        exit(1)
+
+    if not list_files:
+        logger.info("No file/folder found to analyze. Exiting...")
+        exit(1)
+
+    if num_cores > 1:
+        csv_info_raw = Parallel(n_jobs=num_cores)(delayed(analyze_csv)(file_path,
+                                                                    num_rows=num_rows,
+                                                                    date_process=args.date_process,
+                                                                       output_mode=args.output_mode)
+                                              for file_path in tqdm(list_files))
+        csv_info = {file_path.stem: info for (file_path,info) in zip(list_files, csv_info_raw)}
+
+    else:
+        csv_info = {}
+        for file_path in tqdm(list_files):
+            dataset_id = file_path.stems
+            analysis_output = analyze_csv(file_path,
+                                          num_rows=num_rows,
+                                          date_process=args.date_process)
+            csv_info[dataset_id] = analysis_output
+
+    logger.info("Saving info to JSON")
+    # check that we have a good result
+    # _check_full_report(analysis_output, logger)
+
+    # Write your file as json
+    output_folder = Path(args.output_folder)
+    if not output_folder.exists():
+        os.makedirs(output_folder)
+
+    output_file = output_folder / f"{args.date_process}.json"
+    with open(output_file, 'w') as fp:
+        json.dump(csv_info, fp, indent=4, ensure_ascii=False)
```

Comment:
I think this file is not needed in csv-detective python package. I'm not sure we have to keep it here. WDYT @psorianom ?

Reply:
ok deleted !